source/material_model/interface.cc

```diff
@@ -86,7 +86,7 @@ namespace aspect
       (void)dependence;
 
       Assert (viscosity_depends_on(dependence) == false,
-              ExcMessage ("For a model declaring a certain dependence, "
+              ExcMessage ("For a model declaring a certain dependence on viscosity, "
                           "the partial derivatives have to be implemented."));
```

Comment:
You got the dependence the wrong way around. The test does not check that something depends on the viscosity, but that the viscosity depends on something. You can change the text to reflect that it is that way? (Same below.)